Added Org name as ChooseApp in TRV10 folder:

Shaikh-Dilshad opened the pull request:
* Added pramaan flow 1a , flow 1b and flow 2 logs 
* please look these 3 logs i will upload other logs later

Shaikh-Dilshad commented:
@92shreyansh is my logs are verified? but its not showing in the TRV10 folder

Shaikh-Dilshad commented:
1. **Added other pramaan logs TRV10 Flows logs:**

* Added Flow 3-Merchant Side Full Order Cancellation logs
* Added Flow 4-Technical Cancellation (Negative Scenario- TTL is expired, Status , Cancel) logs
* Added Flow 5-Journey Updation (This may include Payment, Fare) logs
* Added Flow 6- Order to Confirm (assign driver post on_confirm) (Self Pickup) logs
* Added Flow 7a-Order to Confirm (driver not found on on_confirm)
* Added flow b


2.  ****Added RSF logs**** 

* flow 1 , flow 2 and flow 3 will be adding others later

sourabhsaini11 commented:
@Shaikh-Dilshad 
common_issues
- on_action calls timestamp should be greater than action call's timestamp
- gps at fulfillment.stops should be precise upto 6 decimal places
- incorrect format for tag SETTLEMENT_WINDOW, value should be P1D instead
- SETTLEMENT_AMOUNT should be calculated by deducting Buyer finder fee from total quote price
- your bap_id is not reflecting when checked at pre-prod env registry. please reupload the logs with correct registered bap_id.

confirm
- missing payment.params

on_confirm
- authorization.stops shouldn't be present when the driver is not assigned at /on_confirm.
- tag DELAY_INTEREST, COURT_JURISDICTION, SETTLEMENT_AMOUNT & SETTLEMENT_WINDOW value mismatch b/w action & on_action calls

on_status
- message_id mismatch against status's message_id

on_cancel(soft)
- missing order.cancellation

Flow-4
cancel
- cancellation_reason_id should be 0 for technical cancellation

Flow-6 & 7
on_search
- missing fulfillments with type Self-Pickup while flow represents the same

Shaikh-Dilshad commented:
@sourabhsaini11 

1. on_action calls timestamp should be greater than action call's timestamp
-> actually we are buyer and in these log seller is pramaan so the on action is from pramaan in that i can't do anything

And on_action is from pramaan saurabh i can't do anything in that other then whatever the issues related to action i will solve that

Shaikh-Dilshad commented:
@sourabhsaini11  below is my lookup api response see we are subscribed in pre-prod

{
        "subscriber_id": "backend.choosecabs.com",
        "status": "SUBSCRIBED",
        "ukId": "66fe660998f25",
        "subscriber_url": "https://backend.choosecabs.com/api/trv10",
        "country": "IND",
        "domain": "ONDC:TRV10",
        "valid_from": "2024-12-11T13:44:54.101Z",
        "valid_until": "2030-06-19T11:57:54.101Z",
        "type": "BAP",
        "signing_public_key": "sgJgGe3xh8pdYoj5NrUuzKhx4h9P//UOY3jWtH5xz/M=",
        "encr_public_key": "MCowBQYDK2VuAyEAN683ZAgZUgv2n3xs8kLovdswU7Wbd39RrEVUBU0pZxE=",
        "created": "2024-12-11T11:26:52.374Z",
        "updated": "2024-12-11T11:26:52.381Z",
        "br_id": "66fe660998f25",
        "city": "std:022"
    }

Shaikh-Dilshad commented:
@sourabhsaini11 
All FLows Modified:
* modified the logs as per review
* as buyer i made changes only on action not on on_action because the on_action is coming from pramaan
* i have added response from lookup and we are subscribed on pre prod please do check

Shaikh-Dilshad commented:
RSF folder modified:
* Added RSF flow 4  Miscellaneous fund movement
* Added RSF flow 5  Nil settlement

Shaikh-Dilshad commented:
@sourabhsaini11 @92shreyansh 
Uploaded the flows logs and RSF logs please do check

sourabhsaini11 commented:
@Shaikh-Dilshad 
common_issues
- on_action calls timestamp should be greater than action call's timestamp
- tags DELAY_INTEREST, COURT_JURISDICTION, SETTLEMENT_AMOUNT value mismatch b/w action on_action calls

Shaikh-Dilshad commented:
@sourabhsaini11 
As i mentioned in previous PR that we are buyer and its seller (pramaan) issue who is sending the on_action . our job is to send only action

Shaikh-Dilshad commented:
@sourabhsaini11 @92shreyansh 
Please do check the logs saurabh in on_action we can't do anything on that this was from pramaan we are buyer only seller is pramaan

92shreyansh commented:
@Shaikh-Dilshad The ask here is not to fix the seller but you should ideally reject such on_actions also looks like you are sending the time as IST timezone the expectation is UTC and not IST.

For the TLC terms these are contract terms if the terms dont match in production ONDC may not be party to resolve a dispute if the TLC are not processed. Please make sure your system does the due validation of these terms as it may impact settlement issues and if terms dont match you should drop the transaction

JyotiSharma0711 commented:
# TRV10/ChooseApp/RSF

## Flow 1

### Settle and On_settle (Common Errors)
context/domain - mismatched in both calls.
context/version - incorrect, expected 2.0.1.

### Report and On_report (Common Errors)
context/domain - wrong in both calls.
context/version - incorrect, expected 2.0.1.

#### On_report
message/settlement/orders/0/provider - additional field, not provided in on_settle call.

---

## Flow 2

### Settle and On_settle (Common Errors)
context/domain - mismatched in both calls.
context/version - incorrect, expected 2.0.1.

#### On_settle
message/settlement/orders/0/error/message - missing object field.
message/settlement/orders/0/id - mismatched with the id in the settle call.

### Report and On_report
context/domain - wrong in both calls.
context/version - incorrect, expected 2.0.1.

#### On_report
message/settlement/orders/0/provider - additional field, not provided in on_settle call.
message/settlement/orders/0/error/message - missing object field.

### Recon
message/orders/0/settlement/0/settlement_ref_no - additional field, status is "Pending".

#### On_recon
message/orders/0/settlement/0/status - missing object field.
message/orders/0/amount/value - mismatched in the settle call.
message/orders/0/settlements/0/commission/value - mismatched in the settle call.

### Recon and On_recon (Common Errors)
context/domain - mismatched in both calls.
context/message_id - mismatched in both calls.
context/timestamp - significant time difference between calls.
context/version - incorrect, expected 2.0.1.

---

## Flow 3

### Settle and On_settle (Common Errors)
context/domain - mismatched in both calls.
context/version - incorrect, expected 2.0.1.

#### On_settle
message/settlement/orders/0/error - missing object field (status is "NOT Settled").
message/settlement/orders/0/error/message - missing object field.

### Report and On_report
context/domain - wrong in both calls.
context/version - incorrect, expected 2.0.1.

#### On_report
message/settlement/orders/0/provider - additional field, not provided in on_settle call.
message/settlement/orders/0/error/message - missing object field.

### Recon
message/orders/0/settlement/0/settlement_ref_no - additional field, status is "Pending".

#### On_recon
message/orders/0/settlement/0/status - missing object field.
message/orders/0/amount/value - mismatched in the settle call.
message/orders/0/settlements/0/with_holding_amount/value - missing.

### Recon and On_recon (Common Errors)
context/domain - mismatched in both calls.
context/message_id - mismatched in both calls.
context/timestamp - significant time difference between calls.
context/version - incorrect, expected 2.0.1.

---

## Flow 4

### Settle and On_settle
context/domain - mismatched in both calls.
context/version - incorrect, expected 2.0.1.
message/collecter_app_id - additional field.
message/receiver_app_id - additional field.
message/settlement/orders/0/collector - additional field.
message/settlement/orders/0/interparticipant - additional field.
message/settlement/orders/0/id - additional field.

---

## Flow 5

### Settle and On_settle
context/domain - mismatched in both calls.
context/version - incorrect, expected 2.0.1.

Shaikh-Dilshad commented:
@92shreyansh @sourabhsaini11 
Uploaded basic flows please do check

Shaikh-Dilshad commented:
IGM2.0 LOGS modified:
* reuploaded the issue_close json

Shaikh-Dilshad commented:
@JyotiSharma0711 
Re Uploaded The RSF logs please do check

Shaikh-Dilshad commented:
@amitsinha07 
iGM2.0 preprod is not completely develop yet so amit from daily dev call told me that issue_status is optional  i can skip that

amitsinha07 commented:
@Shaikh-Dilshad 

You are correct about the `issue_status` and `on_issue_status` calls. However, your flow is incomplete, as the issue cannot be resolved in just one call. Please review the API contract and provide the flow accordingly.

JyotiSharma0711 commented:
# TRV10/ChooseApp/RSF 

## Flow 1

### Settle
message/settlement/orders/0/collector/amount/value - Format should be in ^\d+\.\d{2}$.

### Settle and On_settle
context/version - Incorrect version, expected 2.0.1.
collector_app_id - "collector_app_id should not match with bpp_id" 

### Report and On_report 
context/version - Incorrect version, expected 2.0.1.
**On_report**
  - collector_app_id - "collector_app_id should not match with bpp_id"

---

## Flow 2

### Recon
context/version - Incorrect version, expected 2.0.1.
message/orders/0/settlements/0/status - Must be one of the allowed values: PENDING, SETTLED, TO-BE-INITIATED.

### On_recon
context/version - Incorrect version, expected 2.0.1.
message/orders/due_date - Object field is missing.
message/orders/0/recon_accord - Must be boolean.
message/orders/0/settlements/0/status - Must be one of the allowed values: PENDING, SETTLED, TO-BE-INITIATED.
"Settlement updated_at timestamp cannot be greater than context timestamp."

---

## Flow 3

### Recon
context/version - Incorrect version, expected 2.0.1.
message/orders/0/settlements/0/status - Must be one of the allowed values: PENDING, SETTLED, TO-BE-INITIATED.

### On_recon
context/version - Incorrect version, expected 2.0.1.
message/orders/due_date - Object field is missing.
message/orders/0/recon_accord - Must be boolean.
"Settlement updated_at timestamp cannot be greater than context timestamp."

---

## Flow 4

### Settle and On_settle
context/version - Incorrect version, expected 2.0.1.

---

## Flow 5

### Settle and On_settle
context/version - Incorrect version, expected 2.0.1.

Shaikh-Dilshad commented:
@JyotiSharma0711 
i checked the logs its 2.0.1 version and about the collector_app_id the payment is going to be collect by bpp collector will be bpp then right?

on_recon is sent by mock server

Shaikh-Dilshad commented:
@JyotiSharma0711 
Re Uploaded the RSF logs please do check

JyotiSharma0711 commented:
# TRV10/ChooseApp/RSF (2)

 (common errors)**
**Error**:
  - context/version is wrong, expected version is 2.0.0.

Shaikh-Dilshad commented:
@JyotiSharma0711  you told me again and again that expected version is  2.0.1 kindly look in previous review

Shaikh-Dilshad commented:
@JyotiSharma0711 can i just edit it out the version because my flow is correct right? only the version issue and you are the one who told me in previous review that expected 2.0.1 and its minor changes i will do it in my end please approve the logs

JyotiSharma0711 commented:
Hi @Shaikh-Dilshad due to misinterpretation of log validation message it tends to be mistake on my end Please update the logs again with valid version 2.0.0

Shaikh-Dilshad commented:
Hi @JyotiSharma0711  Updated the logs with the valid version please do check

JyotiSharma0711 commented:
Logs approved for RSF 2.0.0

Shaikh-Dilshad commented:
@sourabhsaini11 @92shreyansh 
its been a week since i have uploaded the logs please do check we need to move forward

amitsinha07 commented:
@Shaikh-Dilshad 

# Feedbacks for IGM

## ISSUE(1)  
- **Context/ttl** - Should be `"P30S"` according to the API contract.  

## ON_ISSUE(2)  
- **Context/core_version** - Object field is missing.  
- **Context/ttl** - Should be `"P30S"` according to the API contract.  
- **Message/issue/actors/2/info/org/name** - Mismatched with `bpp_id`.  
- **Message/issue/actors/2/info/org/name-domain** - Mismatched with `context/domain`.  
- **issue/updated_at** - Missing.  

## ON_ISSUE(3)  
- **Context/core_version** - Object field is missing.  
- **Context/ttl** - Should be `"P30S"` according to the API contract.  
- **Message/issue/actors/2/info/org/name** - Mismatched with `bpp_id`.  
- **Message/issue/actors/2/info/org/name-domain** - Mismatched with `context/domain`.  
- **issue/updated_at** - Missing.  
- **issue/created_at** - Missing.  

## ISSUE(4)  
- **Context/ttl** - Should be `"P30S"` according to the API contract.  
- **Message/issue/created_at** - Mismatched with `Message/issue/created_at` in the “issue” call.  
- **Message/issue/created_at** - Should be less than `issue/updated_at`.  

## ON_ISSUE(5)  
- **Context/core_version** - Object field is missing.  
- **Context/ttl** - Should be `"P30S"` according to the API contract.  
- **Message/issue/actors/2/info/org/name** - Mismatched with `bpp_id`.  
- **Message/issue/actors/2/info/org/name-domain** - Mismatched with `context/domain`.  
- **issue/updated_at** - Missing.  
- **issue/created_at** - Missing.  

## ON_ISSUE(6)  
- **Context/core_version** - Object field is missing.  
- **Context/ttl** - Should be `"P30S"` according to the API contract.  
- **Message/issue/actors/2/info/org/name** - Mismatched with `bpp_id`.  
- **Message/issue/actors/2/info/org/name-domain** - Mismatched with `context/domain`.  
- **issue/updated_at** - Missing.  
- **issue/created_at** - Missing.  

## ISSUE(7)  
- **Context/ttl** - Should be `"P30S"` according to the API contract.  
- **Message/issue/created_at** - Mismatched with `Message/issue/created_at` in the “issue” call.  
- **Message/issue/created_at** - Should be less than `issue/updated_at`.  

## ON_ISSUE(8)  
- **Context/core_version** - Object field is missing.  
- **Context/ttl** - Should be `"P30S"` according to the API contract.  
- **Message/issue/actors/2/info/org/name** - Mismatched with `bpp_id`.  
- **Message/issue/actors/2/info/org/name-domain** - Mismatched with `context/domain`.  
- **issue/updated_at** - Missing.  
- **issue/created_at** - Missing.  

## ISSUE(8)  
- **Context/ttl** - Should be `"P30S"` according to the API contract.  
- **Message/issue/created_at** - Mismatched with `Message/issue/created_at` in the “issue” call.  
- **Message/issue/actions/8/tages/0/list/0/value** - Not in the correct format, should be `"THUMB-UP"`.  
- **Message/issue/created_at** - Should be less than `issue/updated_at`.

Shaikh-Dilshad commented:
@amitsinha07 
I have uploaded the IGM2.0. Please check and review it. As we discussed in the meeting, you asked me to modify my action call based on your review. However, the on_action is coming from the mock server, so I can't make any changes to it

amitsinha07 commented:
@Shaikh-Dilshad 

## ISSUE(1)
- No errors.

## ON_ISSUE(2)
- **Context/ttl** should be `"PT30S"` according to the API contract.  
- **Message/issue/created_at** - object field is missing.  
- **Message/issue/updated_at** - object field is missing.  
- **Message/issue/actors/2/info/org/name** - is mismatched with `bpp_id`.  
- **Message/issue/actors/2/info/org/name-domain** - is mismatched with `context/domain`.  

## ON_ISSUE(3)
- **Context/ttl** should be `"PT30S"` according to the API contract.  
- **Message/issue/created_at** - object field is missing.  
- **Message/issue/updated_at** - object field is missing.  
- **Message/issue/actors/2/info/org/name** - is mismatched with `bpp_id`.  
- **Message/issue/actors/2/info/org/name-domain** - is mismatched with `context/domain`.  

## ISSUE(4)
- **Message/issue/actions/0/updated_at** should match with **message/issue/actions/0/updated_at** of the `"ISSUE"` call.  
- **Message/issue/actions/1/updated_at** should match with **message/issue/actions/1/updated_at** of the `"ON_ISSUE(3)"` call.  
- **Message/issue/actions/2/updated_at** should match with **message/issue/actions/2/updated_at** of the `"ON_ISSUE(3)"` call.  
- **Message/issue/actors/2/info/org/name** - is mismatched with `bpp_id`.  
- **Message/issue/actors/2/info/org/name-domain** - is mismatched with `context/domain`.  

## ON_ISSUE(5)
- **Context/ttl** should be `"P30S"` according to the API contract.  
- **Message/issue/actions/0/updated_at** should match with **message/issue/actions/0/updated_at** of the `"ISSUE"` call.  
- **Message/issue/actions/1/updated_at** should match with **message/issue/actions/1/updated_at** of the `"ON_ISSUE(3)"` call.  
- **Message/issue/actions/2/updated_at** should match with **message/issue/actions/2/updated_at** of the `"ON_ISSUE(3)"` call.  
- **Message/issue/created_at** - object field is missing.  
- **Message/issue/updated_at** - object field is missing.  
- **Message/issue/actors/2/info/org/name** - is mismatched with `bpp_id`.  
- **Message/issue/actors/2/info/org/name-domain** - is mismatched with `context/domain`.  
- **Message/issue/actors/3/info/org/name** - is mismatched with `bpp_id`.  
- **Message/issue/actors/3/info/org/name-domain** - is mismatched with `context/domain`.  

## ON_ISSUE(6)
- Same validation checks as **ON_ISSUE(5)**.  

## ISSUE(7)
- **Message/issue/actions/0-5/updated_at** should match the corresponding **message/issue/actions/** values of the respective `"ISSUE"`, `"ON_ISSUE(3)"`, and `"ON_ISSUE(6)"` calls.  
- **Message/issue/actors/2/info/org/name** - is mismatched with `bpp_id`.  
- **Message/issue/actors/2/info/org/name-domain** - is mismatched with `context/domain`.  

## ON_ISSUE(8)
- Same validation checks as **ON_ISSUE(6)**, plus:  
  - **Message/issue/actions/6/updated_at** should match with **message/issue/actions/6/updated_at** of `"ON_ISSUE(8)"` call.  
  - **Message/issue/actions/7/updated_at** should be **less than or equal to** **message/issue/updated_at**.  

## ISSUE(9)
- Same validation checks as **ON_ISSUE(8)**, ensuring sequential consistency across actions.

Shaikh-Dilshad commented:
@amitsinha07 Submitted the IGM2.0 logs do review and modified the action calls only

amitsinha07 commented:
@Shaikh-Dilshad 

# Feedbacks for IGM

## ON_ISSUE(2)
- **Context/ttl**: should be `PT30S` as per the API contract.  
- **Message/issue/actors/2/info/org/name**: is mismatched with **bpp_id**.  
- **Message/issue/actors/2/info/org/name-domain**: is mismatched with **context/domain**.  

## ON_ISSUE(3)
- **Context/ttl**: should be `PT30S` as per the API contract.  
- **Message/issue/actors/2/info/org/name**: is mismatched with **bpp_id**.  
- **Message/issue/actors/2/info/org/name-domain**: is mismatched with **context/domain**.  

## ISSUE(4)
- **Message/issue/updated_at**: should be less than **message/issue/actions/3/updated_at**.  

## ON_ISSUE(5)
- **Context/ttl**: should be `P30S` as per the API contract.  
- **Message/issue/actors/2/id**: cannot be `null`.  
- **Message/issue/actors/3/info/org/name**: is mismatched with **bpp_id**.  
- **Message/issue/actors/3/info/org/name-domain**: is mismatched with **context/domain**.  

## ON_ISSUE(6)
- **Context/ttl**: should be `P30S` as per the API contract.  
- **Message/issue/actors/2/id**: cannot be `null`.  
- **Message/issue/actors/3/info/org/name**: is mismatched with **bpp_id**.  
- **Message/issue/actors/3/info/org/name-domain**: is mismatched with **context/domain**.  

## ISSUE(7)
- **Message/issue/updated_at**: should be less than **message/issue/actions/6/updated_at**.  

## ON_ISSUE(8)
- **Context/ttl**: should be `P30S` as per the API contract.  
- **Message/issue/actors/2/id**: cannot be `null`.  
- **Message/issue/actors/3/info/org/name**: is mismatched with **bpp_id**.  
- **Message/issue/actors/3/info/org/name-domain**: is mismatched with **context/domain**.  

## ISSUE(9)
- **Message/issue/updated_at**: should be less than **message/issue/actions/7/updated_at**.

Shaikh-Dilshad commented:
@amitsinha07 Re Uploaded the IGM logs please do check

sourabhsaini11 commented:
@Shaikh-Dilshad, transactional-logs are cleared. PR will be merged soon.
